### PR TITLE
[Fix] More comprehensive acquisition of Flink-K8s Ingress public endpoints

### LINKED ISSUE #2553: [Bug] Ingress Metadata cannot obtain the 'field. title. io/publicEndpoints' annotation
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

_No response_

### Scala Version

2.12.x

### StreamPark Version

2.0.0

### Flink Version

1.16.1

### deploy mode

kubernetes-application

### What happened

After I configured Ingress, I found that I could not access the web UI.. After investigation, it is found that in the `IngressController.ingressUrlAddress()` method, the ingress address is obtained by obtaining the `field.cattle.io/publicEndpoints` annotation in the Ingress Metadata. When I debugged at the breakpoint, I found that the annotation did not exist in the Metadata of the ingress I deployed, which caused the failure to obtain the ingress address.

This is related to the `ingress controller` type, not all `ingress controller` types support generating this annotation. Therefore, it may be necessary to change the way to obtain the Ingress address.

### Error Exception

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by senlizishi**
## What changes were proposed in this pull request

Issue Number: close #2553  

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala [hunk 1]
```diff
@@ -122,17 +122,18 @@ object IngressController extends Logger {
...
-          val ingressMeta = metas.head
-          val hostname = ingressMeta.hostname
-          val path = ingressMeta.path
-          logger.info(s"Retrieve flink cluster $clusterId successfully, JobManager Web Interface: https://$hostname$path")
-          s"https://$hostname$path"
-        case None => throw new RuntimeException("[StreamPark] get ingressUrlAddress error.")
-      }
+      // for kubernetes 1.22+
+      val fromV1 = Option(client.network.v1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
+        .map(ingress => ingress.getSpec.getRules.get(0))
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala:127]: We can `import scala.collection.JavaConversions._` and replace `.get(0)` to `.head`

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala [hunk 2]
```diff
@@ -122,17 +122,18 @@ object IngressController extends Logger {
...
+        .map(rule => rule.getHost -> rule.getHttp.getPaths.get(0).getPath)
+      // for kubernetes 1.22-
+      val fromV1beta1 = Option(client.network.v1beta1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala]: `lazy val fromV1beta1 = ....`

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala [hunk 3]
```diff
@@ -122,17 +122,18 @@ object IngressController extends Logger {
...
-      IngressMeta.as(publicEndpoints) match {
-        case Some(metas) =>
-          val ingressMeta = metas.head
-          val hostname = ingressMeta.hostname
-          val path = ingressMeta.path
-          logger.info(s"Retrieve flink cluster $clusterId successfully, JobManager Web Interface: https://$hostname$path")
-          s"https://$hostname$path"
-        case None => throw new RuntimeException("[StreamPark] get ingressUrlAddress error.")
-      }
+      // for kubernetes 1.22+
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala]: This logic should be implemented as a part of the entire `ingressUrlAddress` function , as it has already completed the check for the existence of k8s-ingress for the specified flink identifier.

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala [hunk 4]
```diff
@@ -122,17 +122,18 @@ object IngressController extends Logger {
...
+      val fromV1 = Option(client.network.v1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
+        .map(ingress => ingress.getSpec.getRules.get(0))
+        .map(rule => rule.getHost -> rule.getHttp.getPaths.get(0).getPath)
+      // for kubernetes 1.22-
+      val fromV1beta1 = Option(client.network.v1beta1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
+        .map(ingress => ingress.getSpec.getRules.get(0))
+        .map(rule => rule.getHost -> rule.getHttp.getPaths.get(0).getPath)
+      Try {
+        fromV1.orElse(fromV1beta1)
+          .map { case (host, path) => s"https://$host$path" }.get
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/IngressController.scala]: It is necessary to fallback to the native Flink web interface retrieval method for `.getOrElse(clusterClient.getWebInterfaceURL)`, in the case where there is no corresponding ingress object available through both v1 and v1beta.

**Review by Al-assad — CHANGES_REQUESTED** [review]

**Comment by senlizishi** [comment]
@Al-assad @wolfboys PTAL

**Comment by Al-assad** [comment]
> @Al-assad @wolfboys PTAL

Hi Bin, Please modify this section： 
image

**Comment by senlizishi** [comment]
@Al-assad PTAL1111111111111111111

**Comment by senlizishi** [comment]
@Al-assad PTAL

**Review by Al-assad — CHANGES_REQUESTED** [review]
Oh, You forgot to handle the error, please modify it like this:
```scala

val client = new DefaultKubernetesClient

// for kubernetes 1.22+
lazy val fromV1 = Option(client.network.v1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
  .map(ingress => ingress.getSpec.getRules.get(0))
  .map(rule => rule.getHost -> rule.getHttp.getPaths.get(0).getPath)
// for kubernetes 1.22-
lazy val fromV1beta1 = Option(client.network.v1beta1.ingresses.inNamespace(nameSpace).withName(clusterId).get)
  .map(ingress => ingress.getSpec.getRules.get(0))
  .map(rule => rule.getHost -> rule.getHttp.getPaths.get(0).getPath)

Try(
  fromV1.orElse(fromV1beta1)
    .map { case (host, path) => s"https://$host$path" }
    .getOrElse(clusterClient.getWebInterfaceURL)
).getOrElse(throw new RuntimeException("[StreamPark] get ingressUrlAddress error."))
```

**Comment by senlizishi** [comment]
I'm sorry that I'm not familiar with Scala syntax, which caused me to overlook many details. Thank you for your patient guidance @Al-assad

**Review by wolfboys — APPROVED** [review]
LGTM

**Comment by Al-assad** [comment]
LGTM, Finally 😛

**Review by Al-assad — APPROVED** [review]

**Pull request merged**